pkg/resource/event_bus/delta.go

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package event_bus
+
+import (
+	"bytes"
+	"reflect"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+)
+
+// Hack to avoid import errors during build...
+var (
+	_ = &bytes.Buffer{}
+	_ = &reflect.Method{}
+)
+
+// newResourceDelta returns a new `ackcompare.Delta` used to compare two
+// resources
+func newResourceDelta(
+	a *resource,
+	b *resource,
+) *ackcompare.Delta {
+	delta := ackcompare.NewDelta()
+	if (a == nil && b != nil) ||
+		(a != nil && b == nil) {
+		delta.Add("", a, b)
+		return delta
+	}
+
+	if ackcompare.HasNilDifference(a.ko.Spec.EventSourceName, b.ko.Spec.EventSourceName) {
+		delta.Add("Spec.EventSourceName", a.ko.Spec.EventSourceName, b.ko.Spec.EventSourceName)
+	} else if a.ko.Spec.EventSourceName != nil && b.ko.Spec.EventSourceName != nil {
+		if *a.ko.Spec.EventSourceName != *b.ko.Spec.EventSourceName {
+			delta.Add("Spec.EventSourceName", a.ko.Spec.EventSourceName, b.ko.Spec.EventSourceName)
+		}
+	}
+	if ackcompare.HasNilDifference(a.ko.Spec.Name, b.ko.Spec.Name) {
+		delta.Add("Spec.Name", a.ko.Spec.Name, b.ko.Spec.Name)
+	} else if a.ko.Spec.Name != nil && b.ko.Spec.Name != nil {
+		if *a.ko.Spec.Name != *b.ko.Spec.Name {
+			delta.Add("Spec.Name", a.ko.Spec.Name, b.ko.Spec.Name)
+		}
+	}
+	if !reflect.DeepEqual(a.ko.Spec.Tags, b.ko.Spec.Tags) {
+		delta.Add("Spec.Tags", a.ko.Spec.Tags, b.ko.Spec.Tags)
+	}
```

Comment:
I believe you will need to ignore the `Tags` field for the purpose of comparisons and put in a custom comparator function since `Tags` is a `[]struct{Key,Value}` and `reflect.DeepEqual()` doesn't know how to sort that and therefore differences in sort order (and only sort order) result in an erroneous Delta...

See how the IAM controller handles this [here](https://github.com/aws-controllers-k8s/iam-controller/blob/2f0a8cb2c303d0d5ec8d5133b4b350f20d4752be/generator.yaml#L80-L81) and [here](https://github.com/aws-controllers-k8s/iam-controller/blob/2f0a8cb2c303d0d5ec8d5133b4b350f20d4752be/generator.yaml#L119-L121)

Reply:
Done